### DISCUSSION TITLE
11.0.0 Preview1

### DISCUSSION THREAD

**Discussion by danwalmsley** [AvaloniaUI/Avalonia#8785] [General]
https://dev.to/avalonia/turning-it-up-to-11-34jn

Early list of breaking changes https://github.com/AvaloniaUI/Avalonia/wiki/Breaking-Changes

<hr /><em>This discussion was created from the release <a href='https://github.com/AvaloniaUI/Avalonia/releases/tag/11.0.0-preview1'>11.0.0 Preview1</a>.</em>

**Comment by mike-ward** [comment]
`IImage.Source` is non nullable. Is that intentional?

**Reply by maxkatz6** [reply]
I don't think it's intentional.

**Reply by mike-ward** [reply]
I think it should be nullable. Until the image resource is resolved, the source can be null.

**Comment by timunie** [comment]
⚡ **Breaking change in the Theming:** ⚡

- The themes are now in their own nuget-packages. Remember to install your theme.
  - `Avalonia.Themes.[Simple|Fluent]`
- `DefaultTheme` is now `SimpleTheme`
- ~`Resource-Keys` may have changed their name~

to be continued ...

**Reply by maxkatz6** [reply]
> Resource-Keys may have changed their name

No, but now style setters can override locally set values inside of the control template.

**Reply by HobbesKo** [reply]
Before V11.0.0, in `App.axaml`, I had:
`<StyleInclude Source="avares://Avalonia.Themes.Default/DefaultTheme.xaml"/>`
Obviously that no longer works.

So, I now have:
`<StyleInclude Source="avares://Avalonia.Themes.Simple/SimpleTheme.xaml"/>`
but that just throws an exception:
`No precompiled XAML found for avares://Avalonia.Themes.Simple/SimpleTheme.xaml`.

I have installed the `SimpleTheme`:
`<PackageReference Include="Avalonia.Themes.Simple" Version="11.0.0-preview1" />`.

Any chance someone could tell me where I am going wrong, please.

**Reply by maxkatz6** [reply]
There is no "SimpleTheme.xaml" file there https://github.com/AvaloniaUI/Avalonia/tree/master/src/Avalonia.Themes.Simple

Instead you need to use `<SimpleTheme Mode="Light" />`
StyleInclude is a error prone syntax which is hard to deal when paths are changed

**Reply by HobbesKo** [reply]
Thanks maxkatz6.

My app now runs again, but my datagrids are no longer displayed.

Would you happen to know what I need to do to replace:
`<StyleInclude Source="avares://Avalonia.Controls.DataGrid/Themes/Default.xaml"/>`?

I've tried:
`<StyleInclude Source="avares://Avalonia.Themes.Simple/Controls/DataGrid.xaml"/>`.

I'm looking [here](https://github.com/AvaloniaUI/Avalonia/blob/master/src/Avalonia.Controls.DataGrid/Themes/Simple.xaml) but it's not helping.

**Reply by maxkatz6** [reply]
Yeah, it's 
```
<StyleInclude Source="avares://Avalonia.Controls.DataGrid/Themes/Simple.xaml"/>
```

Hopefully this include won't be necessary when 11.0 will be released.

**Reply by HobbesKo** [reply]
It's OK, thanks, I've got my datagrids back with:
`<StyleInclude Source="avares://Avalonia.Controls.DataGrid/Themes/Simple.xaml"/>`.

**Reply by HobbesKo** [reply]
Ooh, you just beat me to it, but many thanks for the help.

**Reply by MasterMann** [reply]
For people looking how to add the theme programmatically without StyleInclude, you can now do the following (assuming you have `Avalonia.Themes.Simple` as a dependency:

```
var test = new SimpleTheme()
{
    Mode = SimpleThemeMode.Dark
};

this.Styles.Add(test);
```

**Comment by mike-ward** [comment]
As far as I can tell, the color of `<Thumb>` can't be changed using the `Background` or `Foreground` properties.

Can't change the background of a scrollbar as well.

**Comment by Happypig375** [comment]
Why 11.0 instead of 1.0?

**Reply by kekekeks** [reply]
It's our 11th release, so we did the same thing as React (they've jumped from 0.15 to 16.0).

**Comment by mike-ward** [comment]
The `NumericUpDown` control no longer appears to bind to data. 

The markup below is used in a `UserControl` where other controls are binding to the same `DataContext` successfully.  This is a compiled binding. I tried with reflection binding before posting here.

The markup below was working in 10.x

```
  <NumericUpDown
    Width="100"
    Minimum="10"
    Maximum="50"
    Increment="0.1"
    ToolTip.Tip="{StaticResource font-size-title}"
    Value="{Binding FontSize, Converter={StaticResource RoundTo}}" />
```

**Reply by jp2masa** [reply]
Probably related to #5981, although the default converter should handle it I think.

**Reply by maxkatz6** [reply]
Custom converter probably returns non decimal

**Reply by mike-ward** [reply]
Yes, custom converter was returning a double. Problem solved.

**Comment by mike-ward** [comment]
Text wrapping in `RichTextBlock` is not breaking on words.

![image](https://user-images.githubusercontent.com/567927/185817677-e0a57233-9d97-4542-afc6-67417b409168.png)

**Reply by Gillibald** [reply]
```
Need a refresher before House of the Dragon?\n\nHere\u0027s a quick glossary of the characters, houses and settings you need to know.\nhttps://t.co/Yec8zquPDg
```

**Reply by mike-ward** [reply]
Text wrap issue with spaces. "so" on the second line is indented.

![image](https://user-images.githubusercontent.com/567927/185967721-b90e2dd1-2d7f-4121-a328-788902f3949f.png)

Source text
```
"On NBC poll, @mmurraypolitics: \u201CWe asked..\u0027are you so angry and upset that you\u0027d be willing to carry a protest sign for a single day?\u0027 We actually found more Democrats saying yes..than Republicans did\u2026Democrats have caught up on the enthusiasm gap with Republicans.\u201D #AMRstaff"
```

**Reply by Gillibald** [reply]
Is that code pushed somewhere? I can't reproduce the issue with plain text. I need the actual list of inlines to be able to reproduce this.

**Reply by Gillibald** [reply]
This works for me:

```
  <RichTextBlock
    Foreground="Black"
    TextWrapping="Wrap">
    <Run>On NBC poll, </Run>
    <Run Foreground="Blue">@mmurraypolitics:</Run>
    <Run> &#x201C;We asked..&#x0027;</Run>
    <Italic>are you so angry and upset that you&#x0027;d be willing to carry a protest sign for a single day?</Italic><Run>&#x0027; </Run>
    <Run>We actually found more Democrats saying yes..than Republicans did&#x2026;Democrats have caught up on the enthusiasm gap with Republicans.&#x201D; </Run>
    <Run Foreground="Blue">#AMRstaff</Run>
  </RichTextBlock>
```

**Reply by mike-ward** [reply]
https://github.com/mike-ward/Loon

The Inlines are built in code and not markup.

`FlowContentService.cs`
`InlinesService.cs`
`TweetItemText.xaml.cs`

Are the files of interest.

You'll need a twitter account.

You can size the window vertically. It crashes in the `CompositingRenderer` if you resize if horizontally.

I use Rider as the IDE to build it.

Also, uses preview 7 of dotnet.

**Comment by mike-ward** [comment]
Resizing the main window causes a hard crash

```
System.InvalidOperationException: Collection was modified; enumeration operation may not execute.
   at System.Collections.Generic.HashSet`1.Enumerator.MoveNext()
   at Avalonia.Rendering.Composition.CompositingRenderer.Update() in /_/src/Avalonia.Base/Rendering/Composition/CompositingRenderer.cs:line 197
   at Avalonia.Rendering.Composition.CompositingRenderer.Paint(Rect rect) in /_/src/Avalonia.Base/Rendering/Composition/CompositingRenderer.cs:line 250
   at Avalonia.Controls.TopLevel.HandlePaint(Rect rect) in /_/src/Avalonia.Controls/TopLevel.cs:line 375
   at Avalonia.Win32.WindowImpl.AppWndProc(IntPtr hWnd, UInt32 msg, IntPtr wParam, IntPtr lParam) in /_/src/Windows/Avalonia.Win32/WindowImpl.AppWndProc.cs:line 538
   at Avalonia.Win32.WindowImpl.WndProc(IntPtr hWnd, UInt32 msg, IntPtr wParam, IntPtr lParam) in /_/src/Windows/Avalonia.Win32/WindowImpl.WndProc.cs:line 30
```

**Reply by mike-ward** [reply]
Other things I noticed about resizing:

- Crashes only when I size width-wise. Can resize height-wise.
- Happens on Windows and Linux

**Comment by dimohy** [comment]
I don't know much about Avalonia yet. So, I need Avalonia Template, where is the template for 11.0?

**Reply by HobbesKo** [reply]
If you mean the Avalonia-11.0.0-preview1 package, you can install via NuGet.
And, I guess, the theme templates are now also on NuGet.

**Reply by dimohy** [reply]
said wrong sorry.

I was going to talk about Avalonia project template for 11.0-preview1. Or I want to know what changes in an existing project template will compile well. In fact, the more precise need is the project source code itself, compiled for 11.0 Preview 1.

**Reply by dimohy** [reply]
Hmm... I was in too much of a hurry. It was solved.

https://github.com/AvaloniaUI/Avalonia/wiki/Breaking-Changes#010--110

Here I found that I had to add 'Avalonia.Thmes.Fluent' as a separate package. And it compiles and runs.

Thanks

**Reply by maxkatz6** [reply]
11.0 is still in preview, very first preview. So you might expect more smaller breaking changes or bugs.

**Comment by ShrutiJaiswal1494** [comment]
After upgrading my sample app to 11.0.0-preview-1, I am getting some errors. Can someone please help me with these?
I see the dependencies are already added in csproj
```
  <ItemGroup>
	<!--This helps with theme dll-s trimming.
	If you will publish your application in self-contained mode with p:PublishTrimmed=true and it will use Fluent theme Default theme will be trimmed from the output and vice versa.
	https://github.com/AvaloniaUI/Avalonia/issues/5593 -->
	<TrimmableAssembly Include="Avalonia.Themes.Fluent" />
	<TrimmableAssembly Include="Avalonia.Themes.Default" />
  </ItemGroup>
```
<img width="912" alt="image" src="https://user-images.githubusercontent.com/79670853/191865367-c7e2a991-cdee-49c9-8b77-96c86eee7b8b.png">

**Reply by maxkatz6** [reply]
Check breaking changes list from this page https://github.com/AvaloniaUI/Avalonia/wiki/Breaking-Changes
Specifically you need to add Avalonia.Themes.Fluent nuget package.

**Comment by rabbitism** [comment]
When are we going to have preview 2?

**Reply by timunie** [reply]
Hi @rabbitism ,

As often in the open source world, I think it's done when it's done. The team is really working hard on the next steps. But anyway, if you want to try out the bleeding edge features, you can always use nightly builds which is the 11.0 preview. See https://github.com/AvaloniaUI/Avalonia/wiki/Using-nightly-build-feed for information about the usage. 

NOTE: You _may_ get breaking changes if you do so.
